exec/java-exec/src/main/resources/rest/index.ftl

```diff
@@ -17,17 +17,56 @@
   <a href="/queries">back</a><br/>
   <div class="page-header">
   </div>
-  <div class="table-responsive">
-    <table class="table table-hover">
-      <tbody>
-        <#list model as stat>
-          <tr>
-            <td style="border:none;"><b>${stat.getName()}</b></td>
-            <td style="border:none; font-family: Courier;">${stat.getValue()}</td>
-          </tr>
-        </#list>
-      </tbody>
-    </table>
+
+  <#list model.getDrillbits() as drillbit>
+    <#if !drillbit.isVersionMatch()>
+      <div id="message" class="alert alert-danger alert-dismissable">
+        <button type="button" class="close" data-dismiss="alert" aria-hidden="true">&times;</button>
+        <strong>Drillbits in the cluster have different versions.</strong>
+      </div>
+      <#break>
+    </#if>
+  </#list>
+
+  <div class="row">
+    <div class="col-md-6">
+      <h3>General Info</h3>
+      <div class="table-responsive">
+        <table class="table table-hover">
+          <tbody>
+            <#assign props = model.getProps()>
+            <#list props?keys as key>
+              <tr>
+                <td style="border:none;"><b>${key}</b></td>
```

Comment:
I see what we're doing and it is clever. However, it forces us to put the user-visible labels as keys. Since hash maps are unordered, it means that we can't control the order of display of the properties. Two solutions.

1. Design the layout:
```
<tr><th>Version</th> ...
<tr><td>${props.version}</td> ...
```

Or provide two levels of dynamics:
```
class DisplayPair { String key; String label; ... }
List<DisplayPair> list = ...; list.add( "version", "Version" };
...
<#list pairs as pair>
  <th>${pair.label}</th>
...
<#list pairs as pair>
   <td>${model[pair.key]}</td>
```

The first is simpler and is fine: it lets the UI decide which attributes to display, in what form.